pil-analyzer/src/type_inference.rs

```diff
@@ -913,18 +972,19 @@ impl TypeChecker {
         inferred_types: HashMap<String, Type>,
     ) -> Result<HashMap<String, HashMap<String, Type>>, Error> {
         inferred_types.into_iter().map(|(name, inferred_type)| {
-            let (source_ref, declared_type) = self.declared_types[&name].clone();
+            let declared_type = self.declared_types[&name].clone();
```

Comment:
Can you create a function here that abstracts some of this functionality so that you don't need to know what `DeclaredType` is in this function? I.e. already call `simplify_type_vars()` in that function etc.